DrillProcess.py

```diff
@@ -23,215 +19,143 @@
+        await ScienceDevice.publish('DrillBottom', {'SetDutyCycle': int(0)})
+    elif axis > DEADZONE:
+        top_move = 'raising'
+        await ScienceDevice.publish('DrillTop', {'SetDutyCycle': int(duty_cycle)})
+    elif axis < -DEADZONE:
+        top_move = 'lowering'
+        await ScienceDevice.publish('DrillTop', {'SetDutyCycle': int(duty_cycle)})
     else:
-        return 'STOPPED'
+        await ScienceDevice.publish('DrillTop', {'SetDutyCycle': int(0)})
 
-async def bottom_is_moving():
-    if drill.storage.bottom_motor_movement == 1:
-        return 'RAISING'
-    elif drill.storage.bottom_motor_movement == 2:
-        return 'LOWERING'
+    #ScienceDevice.top_motor_movement = top_move
+    log.debug('setting top movement to {}'.format(top_move))
+    await ScienceDevice.sleep(SLEEP_DURATION)
+
+@ScienceDevice.on('*/controller{}/joystick2.'.format(controller_num))
+async def top_motor_movement(joystick2, data):
+    global bot_move
+    axis = data[1]
+
+    if axis is None:
+            return
+    else:
+        duty_cycle = axis * DRILL_RAISE_DUTY_CYCLE
+
+    if -DEADZONE < axis < DEADZONE:
+        bot_move = 'stopped'
+        await ScienceDevice.publish('DrillBottom', {'SetDutyCycle': int(0)})
+    elif axis > DEADZONE:
+        bot_move = 'raising'
+        await ScienceDevice.publish('DrillBottom', {'SetDutyCycle': int(duty_cycle)})
+    elif axis < -DEADZONE:
+        bot_move = 'lowering'
+        await ScienceDevice.publish('DrillBottom', {'SetDutyCycle': int(duty_cycle)})
     else:
-        return 'STOPPED'
+        await ScienceDevice.publish('DrillBottom', {'SetDutyCycle': int(0)})
 
-async def is_spinning():
-    if drill.storage.rotating == True:
-        return 'MOVING'
+    #ScienceDevice.bottom_motor_movement = bot_move
+    log.debug('setting bottom movement to {}'.format(bot_move))
+    await ScienceDevice.sleep(SLEEP_DURATION)
+
+@ScienceDevice.on('*/controller{}/trigger.'.format(controller_num))
+async def set_drill_rotation(trigger, data):
+    global rot_move
+    axis = data
+
+    if axis is None:
+        return
+    else:
+        duty_cycle = axis * ROTATION_SPEED
+
+    if -TRIG_DEADZONE < axis < TRIG_DEADZONE:
+        rot_move = 'stopped'
+        await ScienceDevice.publish('DrillSpin', {'SetRPM': int(duty_cycle)})
+    elif axis > TRIG_DEADZONE:
+        rot_move = 'digging'
+        await ScienceDevice.publish('DrillSpin', {'SetRPM': int(0)})
+    elif axis < -TRIG_DEADZONE:
+        rot_move = 'retracting'
+        await ScienceDevice.publish('DrillSpin', {'SetRPM': int(duty_cycle)})
     else:
-        return 'STOPPED'
+        log.error("motor rotation invalid value {}".format(rot_move))
+        return
 
+    log.debug('Setting rotation movement to {}'.format(rot_move))
+    #ScienceDevice,rotation_direction = rot_move
+    await ScienceDevice.sleep(SLEEP_DURATION)
+
+@ScienceDevice.on('controller{}/{}_down'.format(controller_num, 'buttonB'))
 async def hard_stop():
```

Comment:
This function doesn't actually stop all the motors, it tells the arduino to enable the science experiment.